eth/downloader/queue.go

```diff
@@ -794,6 +794,7 @@ func (q *queue) DeliverHeaders(id string, headers []*types.Header, headerProcCh
 func (q *queue) DeliverBodies(id string, txLists [][]*types.Transaction, uncleLists [][]*types.Header, etxLists [][]*types.Transaction, manifests []types.BlockManifest) (int, error) {
 	q.lock.Lock()
 	defer q.lock.Unlock()
+	nodeCtx := common.NodeLocation.Context()
 	trieHasher := trie.NewStackTrie(nil)
 	validate := func(index int, header *types.Header) error {
```

Comment:
@gameofpointers can we use the body validator here, instead of manually re-implementing all the body validation logic?

Reply:
no because validateBody has more checks than just validating bodies. it check if its pruned ancestor. i.e if you have parents state.

Reply:
Ah yes, good point. That should probably change, but in a different PR. I'll create an issue to refactor this. We shouldn't be writing the same validation logic in so many places.

Reply:
https://github.com/dominant-strategies/go-quai/issues/571